src/bp/cluster.ts

```diff
@@ -0,0 +1,52 @@
+import sdk from 'botpress/sdk'
+import cluster from 'cluster'
+import yn from 'yn'
+
+const debug = DEBUG('cluster')
+
+const msgHandlers: { [messageType: string]: (message: any, worker: cluster.Worker) => void } = {}
+
+/**
+ * The master process handles training and rebooting the server.
```

Comment:
The master and workers are exactly the same processes.  We could run a third process only for training, but that would require minimum 100mb of ram and special logic to redirect requests to the good worker. And it would leave the master unused.

There is room for improvement in the future